ironfish-rust/src/frost_utils/split_spender_key.rs

```diff
@@ -0,0 +1,181 @@
+    pub key_packages: HashMap<Identifier, KeyPackage>,
+    pub public_key_package: PublicKeyPackage,
+}
+
+pub fn split_spender_key(
+    coordinator_sapling_key: SaplingKey,
+    min_signers: u16,
+    max_signers: u16,
+    identifiers: Vec<Identifier>,
+) -> Result<TrustedDealerKeyPackages, IronfishError> {
+    let secret = coordinator_sapling_key
+        .spend_authorizing_key
+        .to_bytes()
+        .to_vec();
+
+    let secret_config = SecretShareConfig {
+        min_signers,
+        max_signers,
+        secret,
+    };
+
+    let identifier_list = IdentifierList::Custom(&identifiers);
+
+    let mut rng: rand::prelude::ThreadRng = thread_rng();
+
+    let (key_packages, public_key_package) =
+        split_secret(&secret_config, identifier_list, &mut rng)?;
+
+    let authorizing_key_bytes = public_key_package.verifying_key().serialize();
+
+    let authorizing_key = Option::from(SubgroupPoint::from_bytes(&authorizing_key_bytes))
+        .ok_or_else(|| IronfishError::new(IronfishErrorKind::InvalidAuthorizingKey))?;
+
+    let proof_generation_key = ProofGenerationKey {
+        ak: authorizing_key,
+        nsk: coordinator_sapling_key.sapling_proof_generation_key().nsk,
+    };
+
+    let nullifier_deriving_key = *PROOF_GENERATION_KEY_GENERATOR
+        * coordinator_sapling_key.sapling_proof_generation_key().nsk;
+
+    let view_key = ViewKey {
+        authorizing_key,
+        nullifier_deriving_key,
+    };
+
+    let incoming_view_key = coordinator_sapling_key.incoming_view_key().clone();
+
+    let outgoing_view_key: OutgoingViewKey = coordinator_sapling_key.outgoing_view_key().clone();
+
+    let public_address = incoming_view_key.public_address();
+
+    Ok(TrustedDealerKeyPackages {
+        verifying_key: authorizing_key_bytes,
+        proof_generation_key,
+        view_key,
+        incoming_view_key,
+        outgoing_view_key,
+        public_address,
+        key_packages,
+        public_key_package,
+    })
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use ironfish_frost::{
+        frost::{frost::keys::reconstruct, JubjubBlake2b512},
+        participant::Secret,
+    };
+
+    #[test]
+    fn test_throws_error_with_mismatch_signer_and_identifiers_lengths() {
+        let mut identifiers = Vec::new();
+
+        for _ in 0..10 {
+            identifiers.push(
+                Secret::random(thread_rng())
+                    .to_identity()
+                    .to_frost_identifier(),
+            );
+        }
+
+        let sapling_key = SaplingKey::generate_key();
+
+        let result = split_spender_key(sapling_key, 5, 11, identifiers.clone());
+        assert!(result.is_err());
+        let err = result.err().unwrap();
+        assert_eq!(err.kind, IronfishErrorKind::Frost);
+        assert!(err.to_string().contains("Incorrect number of identifiers."));
+
+        let sapling_key2 = SaplingKey::generate_key();
+
+        let result = split_spender_key(sapling_key2, 5, 9, identifiers.clone());
```

Comment:
Why do you have the sapling_key2 part of the test? Can't you remove this isn't it basically the same as the above test component?

Reply:
i was checking less than + greater than cases. it gives me the following error when i use the same sapling key twice: `use of moved value: `sapling_key`` should i remove the second case?

Reply:
Oh I see no this is fine. I just missed the fact that you were checking greater vs less than, just add some code comments